### ISSUE TITLE
Warn when assigning value to (some of the) built-in bash variables

### ISSUE THREAD

**Issue by fwc** [koalaman/shellcheck#2559]
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
LINES=3
echo $LINES # prints '3'
touch foo
echo $LINES # prints the no of lines the shell window has
```

or

```sh
RANDOM=3
echo $RANDOM # prints random number
```

or

```sh
PWD=here
echo $PWD # prints 'here'
cd /
echo $PWD # prints '/'
```

#### Here's what shellcheck currently says:

shellcheck does not see any issue with the code.

#### Here's what I wanted or expected to see:

I'd like to see a warning that I'm using a variable name might be overwritten by `bash`.
